src/v3/src/components/PasswordWithConfirmation/PasswordWithConfirmation.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Box, FormHelperText } from '@mui/material';
+import { PasswordInput } from '@okta/odyssey-react-mui';
+import { IdxMessage } from '@okta/okta-auth-js';
+import { h } from 'preact';
+import { useCallback, useEffect, useState } from 'preact/hooks';
+
+import { getMessage } from '../../../../v2/ion/i18nTransformer';
+import { useWidgetContext } from '../../contexts';
+import { useOnChange, useValue } from '../../hooks';
+import {
+  ChangeEvent,
+  FormBag,
+  PasswordWithConfirmationElement,
+  UISchemaElementComponent,
+} from '../../types';
+import { getLabelName } from '../helpers';
+
+enum PasswordType {
+  NEW,
+  CONFIRM,
+}
+
+const PasswordWithConfirmation: UISchemaElementComponent<{
+  uischema: PasswordWithConfirmationElement
+}> = ({ uischema }) => {
+  const { input } = uischema.options;
+  const {
+    name: newPwName,
+    label: newPwLabel,
+    options: {
+      inputMeta: {
+        // @ts-ignore expose type from auth-js
+        messages = {},
+      },
+      attributes,
+    },
+  } = input;
+
+  const { dataSchemaRef } = useWidgetContext();
+  const value = useValue(input);
+  const onChangeHandler = useOnChange(input);
+  // Must use this flag to determine which field contains error
+  const [hasNewPwError, setHasNewPwError] = useState<boolean>(false);
+  const [isNewPwTouched, setIsNewPwTouched] = useState<boolean>(false);
```

Comment:
can we focus this PR on form submission validation only, and handle this state in the following PR?

Reply:
The other Pr is already open (and I am waiting for this to merge to accept and resolve conflicts), I think you are looking to have a HOC somewhat contain this type of state which I can continue looking into after this gets merged and resolve them together. 

But this is to at least match parity for now and drive this feature to completion and refactor (if needed) in the other PR that is open.

Reply:
this will end up with not tested code get merged in, let's be focus in this PR for submission validation only

Reply:
I can add a regression test to prevent this.

Also, will stage and commit my unit test.

Reply:
then what's the purpose of splitting the changes, it's getting together into another mega PR

Reply:
I will remove it and handle it in the next PR, but i will create a test and skip it to make sure it gets covered.